arch/arc/core/cache.c

```diff
@@ -218,12 +218,13 @@ int arch_icache_flush_and_invd_range(void *addr, size_t size)
 
 static int init_dcache(void)
 {
-
-	arch_dcache_enable();
+	if (IS_ENABLED(CONFIG_DCACHE_AUTO_ENABLE)) {
+		arch_dcache_enable();
 
 #if defined(CONFIG_DCACHE_LINE_SIZE_DETECT)
-	init_dcache_line_size();
+		init_dcache_line_size();
 #endif
+	}
```

Comment:
I have no experience in the ARC architecture. Is it OK not to enable the cache? 
I ask because there may be some driver that depends on it or always enables it (like it was mentioned [here](https://github.com/zephyrproject-rtos/zephyr/pull/66140#issuecomment-1850190625) for a different arch).

Reply:
I honestly don't know, I just looked for occasions where the caches were enabled.

Reply:
Probably ok to not enable the data cache, I'm not aware of peripherals that would have a problem with that. Not sure there's much added value from this new feature, though.